packages/app/components/stackbox/ConfirmStackModal.tsx

```diff
@@ -78,6 +99,92 @@ export const ConfirmStackModal = ({
     2
   );
 
+  useEffect(() => {
+    if (allowance && BigInt(allowance) >= rawAmount)
+      setStep(CREATE_STACK_STEPS.create);
+  }, [allowance, rawAmount]);
+
+  useEffect(() => {
+    if (!signer || !address) {
+      return;
+    }
+
+    try {
+      const factoryAddress = getOrderFactoryAddress(chain?.id as ChainId);
+      getERC20Contract(fromToken.address, signer)
+        .allowance(address, factoryAddress)
+        .then((value) => setAllowance(value.toString()));
+    } catch (e) {
+      console.error(e);
+    }
+  }, [signer, address, fromToken.address, chain]);
+
+  const approveFromToken = async () => {
+    if (!signer || !address || !chain) {
+      return;
+    }
+
+    const sellTokenContract = getERC20Contract(fromToken.address, signer);
+
+    try {
+      openModal(ModalId.APPROVE_PROCESSING);
+      const approveFactoryTransaction = await sellTokenContract.approve(
+        getOrderFactoryAddress(chain.id),
+        rawAmount
+      );
+      setApproveTx(approveFactoryTransaction);
+
+      await approveFactoryTransaction.wait();
+
+      setStep(CREATE_STACK_STEPS.create);
+      closeModal(ModalId.APPROVE_PROCESSING);
+    } catch (e) {
+      closeModal(ModalId.APPROVE_PROCESSING);
+      console.error(e);
+    }
+  };
+
+  const timeToUnix = (time: Date) => Math.round(time.getTime() / 1000);
+
+  const createStack = async () => {
+    if (!signer || !address || !chain) {
+      return;
+    }
```

Comment:
```suggestion
    if (!signer || !address || !chain) return;
    
```

Reply:
fixed in [223d187](https://github.com/SwaprHQ/stackly-ui/pull/46/commits/223d187fcbdd254401795e95a9767f1a3fe8a8ec)